### DISCUSSION TITLE
Having trouble getting around reddits 100 post limit

### DISCUSSION THREAD

**Discussion by WARl0-01** [Serene-Arc/bulk-downloader-for-reddit#872] [Q&A]
So I've been searching the web around and I'm super confused on how get past this limit. I'm working with python and my solution was just to have a while loop that would, of course, loop the process until all desired posts were acted upon (unsaved.) However that didn't work, and my next solution was able to get to all posts, but that was because the program never stopped. Using BDFR I can clearly see that it gets around this issue with no problem which is why I'm asking here. I saw some stuff about "after" and "before" stuff but without examples I was pretty confused to how they worked or would work into my code. If you'd like to see what code I'm working with then you can check my public repo "Reddit Unsave Bot." If you do choose to check this out, please ignore the horrendous download method and bad code. Ill attach a snippet of the code that I'm talking about. 
![image](https://github.com/aliparlakci/bulk-downloader-for-reddit/assets/113136419/670b563c-30ac-4330-b15c-6fd6d38d68ea)

My apologies if there is already an existing discussion/answer to this question, I couldn't find anything with 100 post limit or anything similar
I'm also quite the coding newbie if you could dumb down the solution yall used that would be appreciated

**Comment by Serene-Arc** [comment]
There's a 1000 post limit, is that what you're referring to?

**Reply by WARl0-01** [reply]
I've seen some people make issues about this on stack overflow, so I'm unsure if this is a blanket problem or a me problem. 
So when I run my program it starts unsaving posts and then stops after exactly 100 posts. I've tried multiple fixes and each time I can't get it to unsave more then 100. You guys seem to be able to act upon all saved posts, (at least to 1000 which is the limit you assume I'm referring to.) If I want to unsave more then 100 posts I have to run my program multiple times which isnt the end of the world but means I cant use this in a batch script.
To rephrase the question: How does BDFR act upon all saved posts under the hard 1000 post limit?

And some added context. In one of my attempts I tried to check if there were any more posts after the code stopped unsaving which didn't work. My current, at least published code, has a while loop that never ends which does work but as said, doesn't end/stop. From some stuff I've read and from those attempts, I assume this has something do with pages where each page, or section, within Reddit holds 100 posts. The for loop stops because it reaches the end of the page, but the never ending while loop workeds. I assume the program went to the next page.
